File: 05_FitTrack/Javascript/fitTrack_base.js
let X_Accel = 0
let Y_Accel = 0
let Z_Accel = 0
let measuring = false
radio.onDataPacketReceived(({ receivedString: name, receivedNumber: value }) => {
    if (name == "x") {
        X_Accel = value
    } else if (name == "y") {
        // Set the y acceleration to value

    } else if (name == "z") {
        // set the z acceleration to value

        write_data()
    } else if (name == "done") {
        serial.writeLine("done")
    }
})
input.onButtonPressed(Button.AB, () => {
    radio.sendValue("done", 0)
    // Set measuring to false

    // Show a happy face on the display

})
input.onButtonPressed(Button.A, () => {
    // Set measuring to true

    // Show a heart on the display

})
input.onButtonPressed(Button.B, () => {
    // Set measuring to false

    // Show check mark (yes icon) on the display
})
function capture_data() {
    X_Accel = input.acceleration(Dimension.X)
    // Set the y and z accelerations to the y and z dimensions

    radio.sendValue("x", X_Accel)
    // Send y and z values over radio

}
function write_data()  {
    serial.writeLine("" + X_Accel + "," + Y_Accel + "," + Z_Accel)
}
radio.setGroup(1)
radio.setTransmitPower(7)
basic.showIcon(IconNames.Yes)
basic.forever(() => {
    if (measuring) {
        capture_data()
        basic.pause(200)
    }
})

File: 05_FitTrack/Python/fitTrack_base.py
from microbit import *
import radio

radio.on()
radio.config(channel=1)
radio.config(power=7)
x_Accel = 0
# Initialize y and z acceleration

measuring = False
display.show(Image.YES)

def capture_data():
    x_Accel = accelerometer.get_x()
    # Get y and z acceleration

    radio.send(str(x_Accel) + ',' + str(y_Accel) + ',' + str(z_Accel))
    
        
def write_data():
    print(x_Accel, y_Accel, z_Accel, sep=',', end='\n')

while True:
    if button_a.is_pressed() and button_b.is_pressed():
        # Set measuring to False
        
        # Send "done" over radio
        
        # Show a Happy face on the display (display.show(Image.HAPPY))
        
    elif button_a.is_pressed():
        # Set measuring to True

        # Show a heart on the display (Image.HEART)
        
    elif button_b.is_pressed():
        # Set measuring to False

        # Show a checkmark on the display (Image.YES)
        
    incoming = radio.receive()
    if incoming is not None:
        if incoming.startswith('done'):
            display.show(Image.HAPPY)
            print("done")
        else:
            x_Accel, y_Accel, z_Accel = incoming.split(",")
            write_data()
    if measuring:
        capture_data()
    sleep(200)